### ISSUE TITLE
Group member attributing not working with posixgroup schema and ldap auth provider

### ISSUE THREAD

**Issue by schlagmichdoch** [directus/directus#11754]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

When configuring a ldap provider I do want to map certain ldap groups to directus roles.
I have a ldap instance configured with posixAccounts and posixGroups. Directus does not map the groups at all.

Unfortunately directus always compares the AUTH_LDAP_GROUP_ATTRIBUTE to the userDn. As I have a ldap instance configured with posixgroups the groups do not contain member attributes with whole userdns but memberUid Attributes with user uids.
Therefore there should be an option to add the possibility to compare to userUid instead of userDn.

I will create a pr

### To Reproduce

Adding following config to directus dotenv file:

``` env
AUTH_PROVIDERS="ldap"
AUTH_DISABLE_DEFAULT=true

AUTH_LDAP_DRIVER="ldap"
AUTH_LDAP_CLIENT_URL="ldap://127.0.0.1:389"
AUTH_LDAP_BIND_DN="uid=directus.readonly,ou=users,dc=example,dc=com"
AUTH_LDAP_BIND_PASSWORD="PASSWORD"

AUTH_LDAP_USER_DN="ou=users,dc=example,dc=com"
AUTH_LDAP_USER_ATTRIBUTE="uid"
AUTH_LDAP_USER_SCOPE="one"
AUTH_LDAP_MAIL_ATTRIBUTE="mail"

AUTH_LDAP_GROUP_DN="ou=groups,dc=example,dc=com"
AUTH_LDAP_GROUP_ATTRIBUTE="memberUid"
```

Ldap instance running on port 389 with posix structure:

/dc=example,dc=com [localhost]
|-- ou=users
&nbsp;&nbsp;&nbsp;&nbsp;|-- uid=directus.readonly
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|-- objectClass=posixAccount
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|-- objectClass=top
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|-- objectClass=inetOrgPerson
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|-- uid=directus.readonly
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|-- uidNumber=1
|-- ou=groups
&nbsp;&nbsp;&nbsp;&nbsp;|-- cn=directus-admin
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|-- objectClass=posixGroup
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|-- objectClass=top
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|-- cn=directus-admin
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|-- gidNumber=1
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|-- memberUid=direcuts.readonly

### Errors Shown

none, but mapping does not work so message about missing permissions is shown:

![Unbenannt](https://user-images.githubusercontent.com/52242352/154866597-889d6d23-01cd-43db-ad8e-71d0c1607645.PNG)

### What version of Directus are you using?

9.5.2

### What version of Node.js are you using?

16.14.0

### What database are you using?

MySQL Community Server 8.0.27

### What browser are you using?

Firefox

### What operating system are you using?

Debian GNU/Linux 10 (buster)

### How are you deploying Directus?

locally (pm2)

**Comment by aidenfoxx** [comment]
The LDAP provider was built against posixs groups, so they should work. I assume you just need to do a deep search because you're organizing the roles in sub-directories. Just add `AUTH_<PROVIDER>_GROUP_SCOPE="sub"` to your config and you should be good. 

If your role in Directus is called "directus-admin" you should also be using the "cn" attribute in your config.

**Comment by schlagmichdoch** [comment]
My groups just do not allow a "member" attribute. Members are simply attributed by adding their uid to a memberUid Attribute in the group. As the [fetchUserGroups function in directus ldap.ts](https://github.com/directus/directus/blob/c02d5615654e2809109edc6723a4cf00c90b73e0/api/src/auth/drivers/ldap.ts#L188) always compares the complete userDn to the specified `AUTH_LDAP_GROUP_ATTRIBUTE` (which is `memberUid`), it simply does not match anything and returns an empty group array in my case.

[The approach with the `AUTH_LDAP_GROUP_MEMBER_IS_DN` is used by the connector from limesurvey](https://github.com/LimeSurvey/LimeSurvey/blob/b926e7419b3d88226952a7532af670d49977dcd8/application/helpers/ldap_helper.php#L199)
Alternatively we could add a config variable setting corresponding to the connection between groups and users or simply copy nextclouds approach: [if config variable AUTH_LDAP_GROUP_ATTRIBUTE`==`memberUid` use identifier instead of userDN](https://github.com/nextcloud/server/blob/c605ef1f432dcd48ba4b3b8cc82551c234dc4f64/apps/user_ldap/lib/Group_LDAP.php#L140)

**Comment by aidenfoxx** [comment]
@schlagmichdoch I updated my answer while I think you were typing. As I mentioned, I believe the better solution is in the provider to check if the group attribute is "memberUid" and server the users "uid" instead of "dn", as it would solve the issue transparently.

Also, it is totally possible to add the member attribute posix groups, hoever, the simpler solution is to just set up a "groupOfNames" to manage Directus user roles.

**Comment by aidenfoxx** [comment]
Well "posixGroups" are just that. They relate more heavily to a unix environment than to AD. So you have more flexibility and compatibility using the more generic AD definitions like "groupOfNames".

But we still... it would be nice to support the posix stuff better.

**Comment by aidenfoxx** [comment]
I tested this locally against openldap but if I could get confirmation from your side too that would be great @schlagmichdoch! https://github.com/directus/directus/pull/11799

**Comment by schlagmichdoch** [comment]
Can confirm this is working with my configuration: posixgroups and anonymous binding #11750 👍

**Comment by schlagmichdoch** [comment]
This is also fixed by #11799 

thanks for the quick fix! Directus really is awesome! 🥳

**Comment by rijkvanzanten** [comment]
Thanks for the help @schlagmichdoch! ❤️
